### DISCUSSION TITLE
Are transparent windows just not supported on Windows currently?

### DISCUSSION THREAD

**Discussion by axelyoungmakes** [gfx-rs/wgpu#6083] [Q&A]
Hi, I've been going down the rabbit hole of trying to render to a transparent window and seem to just be running in circles at this point. Pretty new to most of this so apologies if I'm missing something obvious.

The issue seems to maybe mostly just be that Windows currently reports `CompositeAlphaMode::Opaque` as the only option on any Nvidia or AMD card. Getting that behaviour on my end, and seems to be replicated by others, such as [here](https://github.com/gfx-rs/wgpu/issues/5661#issuecomment-2106400273), [here](https://github.com/bevyengine/bevy/issues/10929#issuecomment-2242931770), and [here](https://github.com/parasyte/pixels/issues/274#issuecomment-1932997740).

I don't actually see any real issue that is dedicated to this specifically, #5661 is up and also references Windows just not intending to support transparent windows, but isn't actually about the above issue, #5368 is sorta along the same lines and mentions most currently that the issue might be due to windows not reporting surface capability correctly. #3486 seems to maybe be tackling the issue somewhat but honestly I don't fully understand some of it. 

I'm mostly just wondering if the observed issue is actually covered in an existing issue somewhere, or if it's just intended behaviour, and if there's any ways to get around it. Thank you!

**Comment by ThatBigDerp** [comment]
Windows is a bit icky with transparent windows. I haven't personally used wgpu for now, but when I used DirectX in order to get transparent window I had to remove the default window buffer, create a swap chain and associate the swap chain to the window using `VisualLayer` which I think is exclusive to DirectX only. If I'm not wrong wgpu is abstraction over lower-level APIs like DirectX so I'll look around what I can do and probably I'll fork to add it as a feature if I got time

**Reply by axelyoungmakes** [reply]
Yeah, I looked around a little bit more after this post (mostly in the Bevy discord), I think from what I gathered the issue has something to do with NVIDIA and AMD drivers now automatically using a DXGI swapchain compared to something else, and wgpu doesn't currently allow you to choose what swapchain is being used. I think. Again I have a very high level (if any) understanding so what I'm saying could be completely wrong.

wgpu's implementation still seems buggy in some elements though, as also after I wrote this I got behaviour similar to #5661 where I actually managed to get window transparency working on some machines, even *with* the alpha mode being Opaque. What did it for me was for some reason it only did this behaviour when I had the present mode to immediate, which I did not before. Regardless, it only worked on some machines I tried it on, so wasn't a valid workaround for me.

**Reply by ThatBigDerp** [reply]
Well wgpu is aiming at providing cross-platform high-level abstraction so I doubt they'll ever allow you to do *that* much. If you really want transparency you either can submit issue/feature request to the wgpu developers so they do something about this, wait until someone does PR/fork (I may do but 0 promises) or build your own abstraction as you did with your own engine /j

**Reply by axelyoungmakes** [reply]
Yeah, the need for it has actually just passed mostly at this point for me, so not a super high priority thing, just leaving it be for now, might delve into it deeper in the future if it is still an issue. Thank you for your insight though!
